public/js/campaign/visitor.controller.js

```diff
@@ -19,13 +19,30 @@ export default angular.module('helloGov')
             let address = $scope.locDetails.formatted_address.replace(/ /g, '%20');
             let latitude = $scope.locDetails.geometry.location.lat();
             let longitude = $scope.locDetails.geometry.location.lng();
-
+            let userState = null;
+            for (let addressComponent of $scope.locDetails.address_components) {
+                if ((addressComponent.types[0] === "country" && addressComponent.short_name === "PR") || addressComponent.types[0] === "administrative_area_level_1") {
+                    userState = addressComponent.short_name;
+                }
+            }
             $http.get('/locateLegislator', { params: { address: address, latitude: latitude, longitude: longitude, campaignId: $scope.campaign } })
                 .then(function (result) {
                     $scope.repFound = result.data.representativeFound;
                     $scope.repInfo = result.data.representativeInfo;
                     $scope.addrForm = false;
-                    if ($scope.repFound) {
+                    let legislatureLevel = result.data.campaign.legislature_level;
```

Comment:
Why the switch from full names to acros? Is this a DB storage thing or a purely frontend conceit?

Reply:
Got answer IRL: this makes more sense for Google Location API result checking